### PR TITLE
SUP-1314: Test Analytics Integration

### PR DESCRIPTION
**Pull request by james2791**
## PR checklist:
- [x] tests added
- [x] examples of each service action (List, Get etc) added to the relevant `examples/` folder (create if new)
- [x] `CHANGELOG.md` updated with pending release information

### CODE REVIEW

File: buildkite/test_runs.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+type TestRunsService struct {
+	client *Client
+}
+
+type TestRun struct {
+	ID                              *string    `json:"id,omitempty" yaml:"id,omitempty"`
+	URL                             *string    `json:"url,omitempty" yaml:"url,omitempty"`
+	WebURL                          *string    `json:"web_url,omitempty" yaml:"web_url,omitempty"`
+	Branch							*string	   `json:"branch,omitempty" yaml:"branch,omitempty"`
+	CommitSha						*string	   `json:"commit_sha,omitempty" yaml:"commit_sha,omitempty"`
```
Comment (mcncl) [inline buildkite/test_runs.go:18]: *SHA

File: buildkite/test_suites.go [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+//
+// Buildkite API docs: https://buildkite.com/docs/apis/rest-api/analytics/suites
+type TestSuitesService struct {
+	client *Client
+}
+
+type TestSuiteCreate struct {
+	Name       						string 	  `json:"name" yaml:"name"`
+	DefaultBranch                   string    `json:"default_branch,omitempty" yaml:"default_branch,omitempty"`
+	ShowApiToken					bool      `json:"show_api_token,omitempty" yaml:"show_api_token,omitempty"`
```
Comment (mcncl) [inline buildkite/test_suites.go:19]: *API

File: buildkite/test_suites.go [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+	TeamUuids                       []string  `json:"team_ids,omitempty" yaml:"team_ids,omitempty"`
```
Comment (mcncl) [inline buildkite/test_suites.go:20]: UUIDs

**Review by mcncl — CHANGES_REQUESTED** [review]
Has `gofmt -w .` been run over this? The tabbing seems off in a couple of places.

File: buildkite/test_runs.go (hunk 1 above)
Comment (james2791) [reply buildkite/test_runs.go:18]: Adjusted 👍

File: buildkite/test_suites.go (hunk 3 above)
Comment (james2791) [reply buildkite/test_suites.go:20]: Adjusted 👍

File: buildkite/test_suites.go (hunk 2 above)
Comment (james2791) [reply buildkite/test_suites.go:19]: Adjusted 👍

**Review by mcncl — APPROVED** [review]

**Pull request merged**